### PR TITLE
cody web ui: fix cody in web not working locally

### PR DESCRIPTION
**Pull request by mrnugget**
Opening up as draft because I'm not sure about the changes we (@eseliger and I) made here, but it fixed it locally.

## Test plan

- Manual testing of talking to Cody locally

### CODE REVIEW

**Comment by sg-e2e-regression-test-bob** [comment]
## Bundle size report 📦

| Initial size | Total size | Async size | Modules |
| --- | --- | --- | --- |
| 0.00% (0.00 kb) | 0.00% (+0.07 kb) | 0.00% (+0.07 kb) | 0.00% (0) |

Look at the [Statoscope report](https://console.cloud.google.com/storage/browser/_details/sourcegraph_reports/statoscope-reports/mrn+es/cody-web-local/compare-report.html;tab=live_object) for a full comparison between the commits 6f50418 and 17a04b6 or [learn more](https://docs.sourcegraph.com/dev/how-to/testing#bundlesize).


<details>
<summary>Open explanation</summary>

- `Initial size` is the size of the initial bundle (the one that is loaded when you open the page)
- `Total size` is the size of the initial bundle + all the async loaded chunks
- `Async size` is the size of all the async loaded chunks
- `Modules` is the number of modules in the initial bundle
</details>

**Comment by mrnugget** [comment]
Ping @philipp-spiess and @thenamankumar here :)

File: client/web/src/stores/codyChat.ts [hunk 1]
```diff
@@ -166,7 +166,7 @@ export const useChatStoreState = create<CodyChatStore>((set, get): CodyChatStore
 
         try {
             const client = await createClient({
-                config,
+                config: { ...config, customHeaders: window.context.xhrHeaders },
```
Comment (philipp-spiess) [inline client/web/src/stores/codyChat.ts:169]: I think we can remove this from here if we already define `customHeaders` in `config` object below.

**Review by philipp-spiess — APPROVED** [review]
@mrnugget 💯% agree that we need our custom headers here. One inline nit because I think we can revert one line and it would still work. 

Any reason this is in draft state?

**Comment by mrnugget** [comment]
>Any reason this is in draft state?

Only reason was that we weren't sure whether this makes sense or is just a hack on top of a proper fix, so I wasn't sure about it.

**Comment by philipp-spiess** [comment]
Gotcha! I think conceptually this makes a lot of sense though 🙂

File: client/web/src/stores/codyChat.ts (hunk 1 above)
Comment (chwarwick) [reply client/web/src/stores/codyChat.ts]: I removed this and verified that it still completed the request successfully in a case inside Sourcegraph App that consistently failed without it.

**Review by thenamankumar — APPROVED** [review]

File: client/web/src/stores/codyChat.ts (hunk 1 above)
Comment (mrnugget) [reply client/web/src/stores/codyChat.ts]: I also removed it, but ever since the branch doesn't work for me anymore locally. I'll need to try again.

**Pull request merged**